src/components/student/ScheduleView.js

```diff
@@ -9,11 +12,182 @@ import React, {useState} from 'react';
 // issue a DELETE with URL /enrollment/{enrollmentId}
 
 const ScheduleView = (props) => {
-    
-   
+
+    const headers = ['CourseId', 'Title', 'Credits', 'SecId', 'SecNo', 'Building',
+        'Room', 'Times', 'Year', 'Sem', 'ID', 'Email', 'Name', 'Grade', ''];
+
+    // TODO: Change StudentId Once Login is Implemented.
+    const [query, setQuery] = useState({year:'', semester:'', studentId:3});
+
+    const [entries, setEntries] = useState([]);
+
+    const [message, setMessage] = useState('');
+
+    const [isDataFetched, setIsDataFetched] = useState(false);
+
+    const fetchSchedule = async () => {
+
+        // Resets message so messages don't linger longer than they should
+        setMessage('');
+
+        // Check if any search params are empty
+        if (hasEmptyParams()) {
+            setMessage('Please Enter Search Params');
+            return;
+        }
+
+        // Check if search params are valid
+        if (!areParamsValid()) {
+            setMessage('Please Enter Valid Params');
+            return;
+        }
+
+        try {
+            const enrollments = await getEnrollments();
+            setEntries(enrollments);
+            setIsDataFetched(true);
+        } catch (error) {
+            setMessage("Error: " + error);
+        }
+    }
+
+    const hasEmptyParams = () => {
+        return query.year === '' || query.semester === '' || query.studentId === '';
+    }
+
+    const areParamsValid = () => {
```

Comment:
I believe using a regex with .test() would allow for better parameter validation